include/flatbuffers/grpc.h

```diff
@@ -23,50 +23,271 @@
 #include "grpc++/support/byte_buffer.h"
 #include "grpc/byte_buffer_reader.h"
 
+namespace flatbuffers {
 namespace grpc {
 
+enum AddRef { ADD_REF };
+enum StealRef { STEAL_REF };
+
+// Message is a typed wrapper around a buffer that manages the underlying
+// `grpc_slice` and also provides flatbuffers-specific helpers such as `Verify`
+// and `GetRoot`. Since it is backed by a `grpc_slice`, the underlying buffer
+// is refcounted and ownership is be managed automatically.
+template <class T>
+class Message {
+ public:
+  inline Message() : slice_(grpc_empty_slice()) {}
+
+  inline Message(grpc_slice slice, AddRef) : slice_(grpc_slice_ref(slice)) {}
```

Comment:
I based this on a `grpc_slice` wrapper I saw in the gRPC code. One benefit that I could see is that it makes the calls more readable (i.e. you see `ADD_REF` rather than `true`), but could also be done with a single enum type. I suppose making it two distinct types means they can have to completely different impls for the two ctors.

In any case, the ctors here are trivial so I'm happy to make it a bool.